Mathlib/RingTheory/SimpleModule/Isotypic.lean

```diff
@@ -0,0 +1,249 @@
+/-
+Copyright (c) 2025 Junyan Xu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junyan Xu
+-/
+import Mathlib.RingTheory.SimpleModule.Basic
+
+/-!
+# Isotypic modules and isotypic components
```

Comment:
Could you please write a few sentences, e.g. explaining what an isotypic module is? And maybe pointers to the main result(s)?

Reply:
Sorry, missed the review. Module docstrings are added in the subsequent PR (specifically [f5b94a2](https://github.com/leanprover-community/mathlib4/pull/24192/commits/f5b94a2bf5a85422b7349157a454b0d4e3a7e0ae)), since both PRs touch these two files.